contracts/GeneralizedSwapMigrator.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/ISwap.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title GeneralizedSwapMigrator
+ * @notice This contract is responsible for migration liquidity between pools
+ * Users can use this contract to remove their liquidity from the old pools and add them to the new
+ * ones with a single transaction.
+ */
+contract GeneralizedSwapMigrator is Ownable {
+    using SafeERC20 for IERC20;
+
+    struct MigrationData {
+        IERC20 oldPoolLPTokenAddress;
+        IERC20 newPoolLPTokenAddress;
+        IERC20[] underlyingTokens;
+    }
+
+    uint256 private constant MAX_UINT256 = 2**256 - 1;
+    mapping(address => mapping(address => MigrationData)) migrationMap;
```

Comment:
what's the purpose of having a map of a map like this as opposed to a shallow map `oldAddress => mdata`? 

If I'm the frontend, how would I know which of the addresses in the map is the current preferred migration destination? eg
```
log(migrationMap(oldAddr))
{
 0x1: {},
 0x2: {}
}
```

Reply:
Hmm good point. We want to have preferred migration destinations for each LP tokens. I can change this to a shallow map